lib/graphiti/util/transaction_hooks_recorder.rb

```diff
@@ -0,0 +1,46 @@
+module Graphiti
+  module Util
+    class TransactionHooksRecorder
+      class << self
+        def record
+          reset_hooks
+
+          begin
+            transaction_response = yield.tap do |r|
+              run(:before_commit)
+              r[:after_commit_hooks] = hook_set(:after_commit)
```

Comment:
I personally find this less obvious and more indirection than `Hooks.after_commit_hooks`. But it's not really a problem and an internal API we can change later.